synapsecns/sdk
# Merge dev into master

## New Things

- Metis Stableswap pool  (4d0c619c658fe4a0007a67c9761909e6ed77dbae)
- Metis USDC address (788836291c981dd7742302fe51e22c030bd92f79)
- Tests for Metis stableswap pool (e1d144d7e1eeaccea760e1074ecc63be8a9dd8e0)
- NEWO token on Ethereum, Arbitrum, and Avalanche (523ae4fbd4f99ecfaa70b9f588dc3ebf6b7d5ae5, 86d0c59bfa54b433598957bb502fdc7bc3fa106c, 56a7f452ba2736d4951032fc5cf3132a70b6bd78)

## Updated contract addresses

- Metis L2BridgeZap (97e3cb948d56620a846ffd7f8db59b260f81587f)
- BridgeConfigV3 (ad59a5fc4fd25d303c87a0ef6ff7fa14126b5982)

## Fixes/improvements

- Better gas fee/limit population, as well as specific parameters for BSC, Polygon, and Optimism (6a806506239cbbee86f3b36057c1c80b2e2c666a, 609f30ff48e1c7785b873816a3435a38b67cef0b, 3434151a601e10c9673be1cf92ec3b9bb96ae594, 5e98ccde17a946c284f6147e782ce5c193bf028e)

## Test improvements / changes

- "Static call" testing for transaction execution (de68372b6e6b9680d3bf068ce12e6c7ab5fd2ad9)
- Use `[test cases array].forEach(...)` where possible to ensure mocha actually runs all tests and test cases (194d62ec986feda90f0dccd24ee2288a5c506cca)
- Basic sanity checks for token data, which ensures that things like addresses, decimals, and symbols have expected values (or lack thereof) (3f685b8d640ec9a66c55e7ac053a3ea1e9293e6b)
- Add a few extra test cases for Slippages in [test/bridge/Slippages-test.ts](https://github.com/synapsecns/sdk/blob/dev/test/bridge/Slippages-test.ts) (04d4facb54c529d858a2c5f457ace72c2f3b77f9)
- Lower timeouts for tests in [test/bridge/SynapseBridge_Provider_Interactions-test.ts](https://github.com/synapsecns/sdk/blob/dev/test/bridge/SynapseBridge_Provider_Interactions-test.ts) (d85d555d98b8dea7a21c76cb44da413757074675)
- Actually show the entire error message if/when one is thrown by tests (d6eb8e0c6d398a8bc838306e32525d19943896d5)
- Implement basic setup for testing built transaction data (0a5a7077eeafa880018fdfd5275c89ee1f40015c)

## Other changes

- Allow renovate to automerge minor-version dependency updates if they pass tests (e484e8e6b026749f35ea7b837d0104c25e860c8e, db92533dd8886112d39b2342d806309d7ee19adb, 4f2fce61def210ba0f94f1da4fa6ce8e44c879b8)
